results2junit

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python
+#
+# results2junit - convert docker-autotest results file to Jenkins format
+#
+"""
+Convert docker-autotest results file (named 'status') to a results.junit
+file suitable by Jenkins.
+"""
+
+import argparse
+import os
+import re
+import sys
+import time
+
+
+def xml_escape(input_string):
+    """
+    Escape XML-sensitive characters in a string
+    """
+    if not input_string:
+        return ''
+    if isinstance(input_string, basestring):
+        s_out = input_string
+    else:
+        s_out = '%s' % input_string
+
+    s_out = s_out.replace('&', '&amp;')
+    s_out = s_out.replace('<', '&lt;')
+    s_out = s_out.replace('>', '&gt;')
+    s_out = s_out.replace('"', '&quot;')
+    return s_out
+
+
+class TestSuite(object):
+    """
+    Set of test results, initialized from a list read by blah blah
+    """
+
+    def __init__(self, name, results):
+        self.input_name = name
+        # Strip gobbledygook from rhel-latest-7-3-ADEPT-full-26-a451b130bf40
+        self.name = re.sub(r'-\d+-[0-9a-f]{11,}$', '', name)
+        self.results = results
+        self.count = {'tests': 0, 'failures': 0, 'errors': 0, 'skipped': 0}
+
+        # Last result entry is an overall status
+        overall = self.results.pop()
+        if overall['name'] != '----':
+            raise ValueError('Expected "----" as last result; got %s' %
+                             overall['name'])
+        self.time = overall['run_time']
+        parsed_time = time.localtime(overall['timestamp'])
+        self.timestamp = time.strftime('%Y-%m-%d', parsed_time)
+        self.testcases = []
+        for result in self.results:
+            testcase = TestCase(result, self.name)
+            self.add_testcase(testcase)
+
+        self._consolidate_garbage_checks()
+
+    def add_testcase(self, testcase):
+        """
+        Add one testcase to our list. Update relevant counts (total number
+        of tests, number of failures/errors/skipped).
+        """
+        self.testcases.append(testcase)
+
+        self.count['tests'] += 1
+        if testcase.category in self.count:
+            self.count[testcase.category] += 1
+        elif testcase.category:
+            print "********* WEIRD %s" % testcase.category
+
+    def _consolidate_garbage_checks(self):
+        for i, tc in enumerate(self.testcases):
+            if tc.name == 'garbage_check':
+                tc.classname = self.testcases[i-1].classname
+                tc.name = self.testcases[i-1].name + '--garbage-check'
+
+    def write_xml(self, outfile):
+        """
+        Writes a junit XML file.
+        Yes, this is hand-created XML and it's nauseating. That's how
+        the existing autotest JUnit_api.py does it, and that's the quickest
+        way for me to do it right now.
```

Comment:
``ElementTree`` is super easy to work with, especially with all the looping you've got going on here.  It also takes care of all the escaping and encoding stuff for you.  Here's a cheap example:

```python
from xml.etree import ElementTree as ET

tree = ET.ElementTree(ET.Element("testsuites"))
sweet = ET.Element("testsuite", dict(name="foo", timestamp="now", tests="1", errors="2", failures="3", skipped="4"))

props = ET.Element("properties")
for _dict in [dict(foo="bar"), dict(sna="foo")]:
    props.append(ET.Element("property", **_dict))

tc_elements = [ET.Element("testcase", _dict)
               for _dict in [dict(classname="a", name="<b>"),
                             dict(classname="c", name="\"d\"")]]
sweet.append(props)
sweet.extend(tc_elements)
tree.getroot().append(sweet)
# tree.write(filepath, encoding="UTF-8")
ET.dump(tree)
```

The other way to do it is with the ``SubElement()`` method.  Check out the python docs for it, it's ***way*** simpler than what you probably imagine.